closed/src/java.base/share/classes/com/sun/crypto/provider/NativeCipherBlockChaining.java

```diff
@@ -72,6 +76,33 @@
         }
     }
 
+    private static final class CBCCleanerRunnable implements Runnable {
+        private int ctxIndx;
+        private long nativeContext;
+
+        public CBCCleanerRunnable(int ctxIndx, long nativeContext) {
+            this.ctxIndx = ctxIndx;
+            this.nativeContext = nativeContext;
+        }
+
+        @Override
+        public void run() {
+            /*
+            * Release the CBC context.
+            */
+            if(ctxIndx == -1) {
```

Comment:
This isn't going to work. As currently written, ctxIndx here is a copy of the value in the NativeCipherBlockChaining instance, and doesn't get updated.

Reply:
@pshipton I don't understand your comment - that's the intention that ctxIndx will be a copy of the value in the NativeCipherBlockChaining instance. ctxIndx is assigned to the NativeCipherBlockChaining instance in the constructor (in the call getContext()) and doesn't change for the lifetime of the instance, note that only getContext() changes ctxIndx. At cleaning time, if the ctxIndx value is -1 a special release should be taken. otherwise the value should be returned to the global object avStack.

Reply:
Ok, I see it now.
For consistency, please add a space, it should if `if (`